core-java/src/main/java/org/spine3/server/procman/ProcessManager.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2015, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.procman;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.protobuf.Message;
+import com.google.protobuf.Timestamp;
+import org.spine3.base.*;
+import org.spine3.internal.EventHandlerMethod;
+import org.spine3.server.Entity;
+import org.spine3.server.internal.CommandHandlerMethod;
+import org.spine3.util.Classes;
+import org.spine3.util.Events;
+import org.spine3.util.MethodMap;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nullable;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A central processing unit used to maintain the state of the sequence and determine the next processing step
+ * based on intermediate results.
+ * <p>
+ * An independent component that reacts to domain events in a cross-aggregate, eventually consistent manner.
+ * Used for externalizing the decisions on the logic flow from the business logic.
+ * <p>
+ * Event/command handlers are invoked by the {@link ProcessManagerRepository}
+ * that manages instances of a process manager class.
+ * <p>
+ * There is a common confusion between Process Managers and Sagas.
```

Comment:
We need to make this part less mentory. How about this?

For more information on Process Managers, please see:
<ul>
<li><a href="http://www.enterpriseintegrationpatterns.com/patterns/messaging/ProcessManager.html">Process Manager Pattern</a></li>
<li>Difference between Process Manager and Saga: <a href="http://kellabyte.com/2012/05/30/clarifying-the-saga-pattern/">Clarifying the saga pattern</a></li>
<li><a href="https://dzone.com/articles/are-sagas-and-workflows-same-t">Are sagas and workflows the same ...</a></li>
<li><a href="https://msdn.microsoft.com/en-us/library/jj591569.aspx">CQRS Journey Guide: A Saga on Sagas</a></li>
</ul>